client/lib/url/index.js

```diff
@@ -180,7 +180,41 @@ function omitUrlParams( url, paramsToOmit ) {
 	return formatUrl( parsed );
 }
 
+/**
+ * Wrap decodeURI in a try / catch block to prevent `URIError` on invalid input
+ * @param  {String} encodedURI URI to attempt to decode
+ * @return {String}            Decoded URI (or passed in value on error)
+ */
+function decodeURIIfValid( encodedURI ) {
+	if ( ! ( isString( encodedURI ) || has( encodedURI, 'toString' ) ) ) {
+		throw new Error( 'Invalid input' );
```

Comment:
What's the purpose for throwing here? We don't appear to be handling this case in the selector.

Reply:
@gwwar the thought was to prevent unexpected behavior when providing a non-string to `decodeURI` & `decodeURIComponent`.  For example: 
```
> x = {};
> decodeURIComponent(x);
"[object Object]"
> y = null;
> decodeURI( y );
"null"
```

Since there's no type checking & no error thrown, invalid usage will be potentially masked.

You're right, though. The case isn't checked for in the calling selector (where `postSlug` has the potential to be `undefined`). 

What if, instead of throwing, these helpers just return an empty string on non-string input?

Reply:
Thanks for the context @jblz! Your suggestion sounds reasonable to me.